File: python/samples/node_proxy/node_proxy.py
import bloock

from bloock.client.entity.network import Network

from utils.sample import Sample
from utils.config import Config


def set_provider(_: Config):
    # we can change the provider by specifing the network and the new url of the provider
    bloock.set_provider(Network.ETHEREUM_MAINNET, "https://eth.someprovider.com")


def set_contract_address(_: Config):
    # we can change the contract address by specifing the network and the new contract address
    bloock.set_contract_address(Network.ETHEREUM_MAINNET, "some contract address")


Sample("set_provider", set_provider)
Sample("set_contract_address", set_contract_address)

File: go/samples/node_proxy/node_proxy.go
package main

import (
	"github.com/bloock/bloock-sdk-go/v2/client"
	"github.com/bloock/bloock-sdk-go/v2/client/entity"
	"github.com/bloock/bloock-sdk-quickstart/utils"
)

func main() {
	utils.Sample("set provider", func(c utils.Config) error {
		// we can change the provider by specifing the network and the new url of the provider
		client.SetProvider(entity.ListOfNetworks().EthereumMainnet, "https://eth.someprovider.com")
		return nil
	})

	utils.Sample("set contract address", func(c utils.Config) error {
		// we can change the contract address by specifing the network and the new contract address
		client.SetContractAddess(entity.ListOfNetworks().EthereumMainnet, "some contract address")
		return nil
	})
}
